### DISCUSSION TITLE
exclude all system dependency

### DISCUSSION THREAD

**Discussion by x1y9** [marcelotduarte/cx_Freeze#2264] [Q&A]
I want to convert my scripts to exe, but without all system dependency like numpy, pytorch etc.
My purpose is just to obstacle my script. So how to do that?

**Comment by marcelotduarte** [comment]
NumPy is not a system dependency. It may be if you are using Anaconda.
The best way is to use a virtual environment [1], as you install only the packages you need and consequently, cx_freeze will only use them when generating the executable.
[1] See the first note in [installation](https://cx-freeze.readthedocs.io/en/stable/installation.html).

That is, if you are using pip, you can create a venv or if you are using conda, use conda create.

And in some cases, you can use [excludes](https://cx-freeze.readthedocs.io/en/stable/setup_script.html#cmdoption-arg-excludes).

**Comment by x1y9** [comment]
The dependency is very large (numpy,scipy,pyTorch...), so I just want to convert my code to exe, then execute on client machine using there Python environment, then I can distribute a small package to client machine.

I know excludes, should I write every 3rd package in excludes?

**Reply by marcelotduarte** [reply]
Yes.

Do you have numpy,scipy,etc as dependency of your code? If so, if you use excludes=["numpy", "scipy"] will broke the code. If not, it works, reducing the files to distribute, but the best you can do it use a venv.
